src/main.rs

```diff
@@ -1,4 +1,143 @@
+    V1,
+}
+
+#[derive(Debug)]
+struct JsonProxyError(reqwest::Error);
+
+#[tokio::main]
+async fn main() {
+    let meta_routes = Router::new()
+        .route("/", get(root_handler))
+        .route("/ping", get(heartbeat_handler));
+
+    let transit_routes = Router::new()
+        .route("/bus", get(transit_bus_handler))
+        .route("/modes", get(transit_modes_handler));
+
+    let api_routes = Router::new()
+        .route("/contacts", get(contacts_handler))
+        .route("/dictionary", get(dictionary_handler))
+        .route("/faqs", get(faqs_handler))
+        .route("/food/named/menu/pause", get(pause_menu_handler))
+        .route("/printing/color-printers", get(color_printers_handler))
+        .route("/spaces/hours", get(hours_handler))
+        .route("/tools/help", get(help_handler))
+        .nest("/transit", transit_routes)
+        .route("/webcams", get(webcams_handler));
+
+    let app = Router::new()
+        .nest("/", meta_routes)
+        .nest("/api/:version", api_routes)
+        .fallback(fallback.into_service());
+
+    Server::bind(&"0.0.0.0:3000".parse().unwrap())
+        .serve(app.into_make_service())
+        .await
+        .expect("server failed to exit successfully");
+}
+
+#[allow(clippy::unused_async)]
+async fn root_handler() -> Result<&'static str, StatusCode> {
+    Ok("Hello world!")
+}
+
+#[allow(clippy::unused_async)]
+async fn heartbeat_handler() -> Result<&'static str, StatusCode> {
+    Ok("pong")
+}
+
+#[allow(clippy::unused_async)]
+async fn fallback() -> impl IntoResponse {
+    (StatusCode::NOT_FOUND, "Not Found".to_string())
+}
+
+async fn gh_pages_handler(filename: &str) -> Result<Json<Value>, JsonProxyError> {
+    let url = format!("https://stodevx.github.io/AAO-React-Native/{}", filename).to_string();
+    let resp = request_handler(&url).await?;
+    Ok(resp)
 }
 
+macro_rules! gh_pages_handler {
+    ($name:ident,$filename:literal) => {
+        async fn $name(_version: Version) -> Result<Json<Value>, JsonProxyError> {
+            let data = gh_pages_handler($filename).await.unwrap();
+            Ok(data)
+        }
+    };
+}
+
+macro_rules! gh_pages_handlers {
+    ($([$name:ident, $filename:literal]),+ $(,)?) => {
+        $(
+            gh_pages_handler!($name, $filename);
+        )+
+    };
+}
+
+gh_pages_handlers!(
+    [contacts_handler, "contact-info.json"],
+    [dictionary_handler, "dictionary.json"],
+    [faqs_handler, "faqs.json"],
+    [pause_menu_handler, "pause-menu.json"],
+    [color_printers_handler, "color-printers.json"],
+    [hours_handler, "building-hours.json"],
+    [help_handler, "help.json"],
+    [transit_bus_handler, "bus-times.json"],
+    [transit_modes_handler, "transportation.json"],
+    [webcams_handler, "webcams.json"],
+);
+
+async fn request_handler(path: &str) -> Result<Json<Value>, JsonProxyError> {
+    let response = reqwest::get(path).await.map_err(JsonProxyError)?;
+
+    response.json().await.map(Json).map_err(JsonProxyError)
+}
```

Comment:
Okay, so the next thing I'd suggest is adding generics to this.

Right now, `request_handler` returns a `Result<Json<Value>, _>`, which is a concrete type.

As it so happens, `serde_json::Value` is a particular type that implements `serde::DeserializeOwned`.  This means it's a type that can be deserialized fully _from any lifetime_, (i.e. without borrowing anything from the internal deserializer).  I'd expect many custom types to behave this way — as long as we can turn a JSON string into a full struct (by cloning / allocating Strings as needed), we can reasonably just assume that we're dealing with `serde::DeserializeOwned` types instead.

What this means is that `request_handler` can be slightly adjusted to a _generic_ function as such:

```rust
async fn request_handler<T>(path: &str) -> Result<Json<T>, JsonProxyError>
where
    T: DeserializeOwned,
{
    let response = reqwest::get(path).await.map_err(JsonProxyError)?;

    response.json().await.map(Json).map_err(JsonProxyError)
}
```

The body actually doesn't really need to change because Rust is smart enough to figure out that the `T` will come into play with the `response.json()` call.

In order to do this, you'll need to add `serde` to the `Cargo.toml` so you get access to the `serde::DeserializeOwned` type. Cargo is smart enough to re-use the already-downloaded `serde` that you downloaded when you installed `serde_json`.

```toml
serde = "1.0.136"
```

Reply:
A small note about how this will change things under the hood, too. If you use a generic, Rust [monomorphizes](https://rustc-dev-guide.rust-lang.org/backend/monomorph.html) the function for each different instance. In this case, we'd get one copy of `request_handler` for a `ContactInfoResponse` type, one copy for e.g. a `DictionaryResponse` type, etc.

There isn't necessarily a _problem_ with that, but you should just be aware that using generics will lead to more copies of functions being generated. It should be possible to use some kind of dynamic return type instead, but I don't think it's worth the effort at this point.